### PR TITLE
Add api-key=dotcom query string parameter when querying the discussionApi

### PR DESCRIPTION
**Pull request by TBonnin**

### CODE REVIEW

**Comment by TBonnin** [comment]
Q: to add test data I removed the `data/discussion` content and rerun the tests locally. Only 3 new test files have been recreated where there were 12 before. Were they stale test data?

File: discussion/app/discussion/DiscussionApi.scala [hunk 1]
```diff
@@ -34,21 +40,24 @@ trait DiscussionApi extends Http with ExecutionContexts with Logging {
...
-                    |${if(params.displayThreaded) "" else "&displayThreaded=false"}
-                    |&showSwitches=true""".stripMargin.replace("\n", "")+
-                    params.maxResponses.map{i => "&maxResponses="+ i}.getOrElse("")
+    val parameters = List(
+      "pageSize" -> params.pageSize,
+      "page" -> params.page,
+      "orderBy" -> params.orderBy,
+      "displayThreaded" -> (params.displayThreaded match {
+        case false => "false"
+        case _ => None}),
```
Comment (jennysivapalan) [inline discussion/app/discussion/DiscussionApi.scala]: Will this evaluate to `displayThreaded = "" ` in the `_` case?

File: discussion/app/discussion/DiscussionApi.scala [hunk 2]
```diff
@@ -149,8 +176,10 @@ trait DiscussionApi extends Http with ExecutionContexts with Logging {
     }
   }
 
-  override protected def getJsonOrError(url: String, onError: (WSResponse) => String, headers: (String, String)*) =
+  override protected def getJsonOrError(url: String, onError: (WSResponse) => String, headers: (String, String)*) = {
+    println(url)
```
Comment (jennysivapalan) [inline discussion/app/discussion/DiscussionApi.scala]: Can remove the `println`

**Comment by jennysivapalan** [comment]
Looks good :+1: 
Yes they were probably stale - they are used in teamcity mainly. I think we only recreate new ones when tests are run locally but perhaps Discussion hasn't been run for a while.

File: discussion/app/discussion/DiscussionApi.scala (hunk 1 above)
Comment (TBonnin) [inline discussion/app/discussion/DiscussionApi.scala]: No. DisplayThreaded won't be added to the query string if None. That's the default behavior of the NetAPorter.uri when adding query string parameters lib https://github.com/NET-A-PORTER/scala-uri/blob/3c438f082392a722e1c0913b69fe21f458de5af2/src/main/scala/com/netaporter/uri/Uri.scala#L65

File: discussion/app/discussion/DiscussionApi.scala (hunk 2 above)
Comment (TBonnin) [inline discussion/app/discussion/DiscussionApi.scala]: oups good catch. thank you

**Comment by rich-nguyen** [comment]
:+1:

**Pull request merged**